### PR TITLE
Adding login flow with CLI

### PR DESCRIPTION
**Pull request by pashakondratyev**
Still need to store the token somewhere safe/sane so that we can re-use it, but this should unblock some flows

### CODE REVIEW

File: packages/cli/src/commands/auth/CommonAuthArgs.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { CliCommonArgs } from "../../CliCommonArgs.js";
+
+export interface CommonAuthArgs extends CliCommonArgs {
+  baseUrl: string;
```
Comment (mfedderly) [inline packages/cli/src/commands/auth/CommonAuthArgs.ts:20]: We should probably be more consistent here with baseUrl vs stack

File: packages/cli/src/commands/auth/login/loginFlow.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    resolve = _res;
+  });
+
+  const server = createServer((req, res) => {
+    const query = parse(req.url!, true).query;
+    res.end("Authenticated");
+    resolve(query["code"] as string);
+  });
+
+  server.listen(port);
```
Comment (mfedderly) [inline packages/cli/src/commands/auth/login/loginFlow.ts:40]: Probably want to give a clean error message if the port is already in use by the app's webpack or whatever

File: packages/cli/src/commands/auth/login/loginFlow.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+  consola.success(`Successfully authenticated!`);
+  return token;
+}
+
+function generateRandomString() {
+  const array = crypto.getRandomValues(new Uint32Array(28));
+  return Array.from(array, dec => {
+    const decimalString = "0" + dec.toString(16);
+    return decimalString.substring(decimalString.length - 2);
```
Comment (mfedderly) [inline packages/cli/src/commands/auth/login/loginFlow.ts:82]: Took me a sec to figure this out. Also is a length of 28 part of a spec somewhere?

```suggestion
    return dec.toString(16).padStart(2,"0");
```

**Review by mfedderly — APPROVED** [review]

File: packages/cli/src/commands/auth/CommonAuthArgs.ts (hunk 1 above)
Comment (pashakondratyev) [reply packages/cli/src/commands/auth/CommonAuthArgs.ts:20]: I borrowed this from the `upload site` command but agreed

**Pull request merged**